glue/qt/widgets/data_slice_widget.py

```
@@ -45,33 +46,54 @@ def __init__(self, label='', pix2world=None, lo=0, hi=10,
-        slider.setMinimum(lo)
-        slider_lbl = QLabel()
-        slider.setMaximum(hi)
-        slider.setValue((lo + hi) / 2)
-        slider.valueChanged.connect(lambda x:
-                                    self.slice_changed.emit(self.mode))
-        slider.valueChanged.connect(lambda x: slider_lbl.setText(str(x)))
-        layout.addWidget(slider_lbl)
+        slider = load_ui('cube_slider')
+        slider.slider
+
+        slider.slider.setMinimum(lo)
+        slider.slider.setMaximum(hi)
+        slider.slider.setValue((lo + hi) / 2)
+        slider.slider.valueChanged.connect(lambda x:
+                                                  self.slice_changed.emit(self.mode))
+        slider.slider.valueChanged.connect(lambda x: slider.label.setText(str(x)))
+
+        slider.label.setMinimumWidth(50)
+        slider.label.setText(str(slider.slider.value()))
+        slider.label.textChanged.connect(lambda x: slider.slider.setValue(int(x)))
+
+        slider.first.clicked.connect(nonpartial(self._browse_slice, 'first'))
+        slider.prev.clicked.connect(nonpartial(self._browse_slice, 'prev'))
+        slider.next.clicked.connect(nonpartial(self._browse_slice, 'next'))
+        slider.last.clicked.connect(nonpartial(self._browse_slice, 'last'))
+
         layout.addWidget(slider)
 
         self.setLayout(layout)
 
         self._ui_label = label
         self._ui_slider = slider
-        self._slider_lbl = slider_lbl
         self._ui_mode = mode
         self._update_mode()
         self._frozen = False
 
+    def _browse_slice(self, action):
+        imin = self._ui_slider.slider.minimum()
+        imax = self._ui_slider.slider.maximum()
+        value = self._ui_slider.slider.value()
+        if action == 'first':
+            value = imin
+        elif action == 'last':
+            value = imax
+        elif action == 'prev':
+            value = max(value - 1, imin)
+        elif action == 'next':
+            value = min(value + 1, imax)
```

nit: do you want to raise an error for other values of 'action'?